lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2727,50 +2729,52 @@ LogicalResult FIRRTLLowering::lowerVerificationStatement(AOpTy op,
   auto predicate = getLoweredValue(op.predicate());
   if (!clock || !enable || !predicate)
     return failure();
+  predicate = builder.createOrFold<comb::AndOp>(enable, predicate);
 
-  addToAlwaysBlock(clock, [&]() {
-    addIfProceduralBlock(enable, [&]() {
-      // Create BOpTy inside the always/if.
-      StringAttr label;
-      if (op.nameAttr())
-        label = op.nameAttr();
-      else
-        label = builder.getStringAttr("");
-      auto svOp = builder.create<BOpTy>(predicate, label);
-      auto annoSet = AnnotationSet(circuitState.circuitOp);
-      StringRef fileName, dir;
-      if (auto a = annoSet.getAnnotation(annoClass)) {
-        fileName = a.getAs<StringAttr>("filename").getValue();
-        dir = a.getAs<StringAttr>("directory").getValue();
-      }
-      if (!fileName.empty() || !dir.empty())
-        svOp->setAttr("output_file",
-                      hw::OutputFileAttr::get(builder.getStringAttr(dir),
-                                              builder.getStringAttr(fileName),
-                                              builder.getBoolAttr(true),
-                                              builder.getBoolAttr(true),
-                                              svOp.getContext()));
-    });
-  });
+  // Create BOpTy inside the always/if.
+  StringAttr label;
+  if (op.nameAttr())
+    label = op.nameAttr();
+  else
+    label = builder.getStringAttr("");
```

Comment:
It looks like the name attributes are all required.  Is the if/else here needed?